src/qtt/measurements/scans.py

```
@@ -1948,6 +1946,21 @@ def acquire_segments(station, parameters, average=True, mV_range=2000,
 
     return alldata
 
+def _is_m4i(instrument):
+    try:
+        is_m4i = isinstance(instrument, qcodes.instrument_drivers.Spectrum.M4i.M4i)
+        if is_m4i:
+               warnings.warn('please use qcodes_contrib_drivers.drivers.Spectrum.M4i instead of qcodes.instrument_drivers.Spectrum.M4i.M4i)')
+               return is_m4i
+    except:
+        is_m4i = False
+    try:
+        import qcodes_contrib_drivers.drivers.Spectrum.M4i
+        is_m4i = isinstance(instrument, qcodes_contrib_drivers.drivers.Spectrum.M4i.M4i)
+    except:
```

also here again a bare Except, catch ImportError, TypeError when needed.